src/unittest/python/test_text_verify.py

```diff
@@ -0,0 +1,290 @@
+
+
+def test_verify_default_full_match():
+    """
+    Tests the return value of the verify function with default args.
+    In this case we expect a 100% match.
+    """
+    verifier = TextVerify()
+    extracted_info = {
+        'identity_number': '7101135111011',
+        'surname': 'Doe',
+        'names': 'John-Michael Robert',
+        'sex': 'M',
+        'date_of_birth': '71-01-13',
+        'country_of_birth': 'RSA',
+        'status': 'Citizen',
+        'nationality': 'RSA'
+    }
+    verifier_info = {
+        'identity_number': '7101135111011',
+        'surname': 'Doe',
+        'names': 'John-Michael Robert',
+        'sex': 'M',
+        'date_of_birth': '71-01-13',
+        'country_of_birth': 'RSA',
+        'status': 'Citizen',
+        'nationality': 'RSA'
+    }
+    assert verifier.verify(extracted_info, verifier_info) == (True, 100.0)
+
+
+def test_verify_threshold():
+    """
+    Tests the return value of the verify function with a specified threshold arg.
+    """
+    verifier = TextVerify()
+    extracted_info = {
+        'identity_number': '7101135111011',
+        'surname': 'Doe',
+        'names': 'John-Michael Robert',
+        'sex': '####################################',
+        'date_of_birth': '71-01-13',
+        'country_of_birth': 'RSA',
+        'status': 'Citizen',
+        'nationality': 'RSA'
+    }
+    verifier_info = {
+        'identity_number': '7101135111011',
+        'surname': 'Doe',
+        'names': 'John-Michael Robert',
+        'sex': 'M',
+        'date_of_birth': '71-01-13',
+        'country_of_birth': 'RSA',
+        'status': 'Citizen',
+        'nationality': 'RSA'
+    }
+    assert verifier.verify(extracted_info, verifier_info, threshold=0.9) == (False, 87.5)
+
+
+def test_verify_min_matches():
+    """
+    Tests the return value of the verify function with a specified minimum number of matches arg.
+    """
+    verifier = TextVerify()
+    extracted_info = {
+        'identity_number': '7101135111011',
+        'surname': 'Doe',
+        'names': 'John-Michael Robert',
+        'sex': 'M',
+        'not valid': '####################################',
+        'not legit': '####################################',
+        'not gonna work': '####################################',
+        'try again': '####################################'
+    }
+    verifier_info = {
+        'identity_number': '7101135111011',
+        'surname': 'Doe',
+        'names': 'John-Michael Robert',
+        'sex': 'M',
+        'date_of_birth': '71-01-13',
+        'country_of_birth': 'RSA',
+        'status': 'Citizen',
+        'nationality': 'RSA'
+    }
+    assert verifier.verify(extracted_info, verifier_info, min_matches=6) == (False, 0.0)
+
+
+def test_verify_verbose():
+    """
+    Tests the return value of the verify function with a specified verbose arg.
+    """
+    verifier = TextVerify()
+    extracted_info = {
```

Comment:
Maybe change the data so that you can clearly see in the result that the match percentage differs between fields, i.e. change some of the values of `extracted_info` and check that the result is still correct.